### ISSUE TITLE
About your implementation

### ISSUE THREAD

**Issue by moonman925** [mindslab-ai/hififace#2]
Hi, thx for sharing your codes, great works!
I have serval questions about the implementatation.
1. according to the issue from the official hififace github repo, both masks are activated by sigmoid but I noticed that you used tanh.
2. did you encounter with situation like, the mask learns a all-zero mask, which cause the model generate the exact same image as target image ?

**Comment by usingcolor** [comment]
Hi! Thank you for your interest!
1. Are you talking about [this issue](https://github.com/johannwyh/HifiFace/issues/3#issuecomment-917888760) on the official project Github? Honestly, I didn’t know this existed.  I decided on some details that were not mentioned on the paper, like activation for masks. I don't think this makes a very big difference during training.

2. That shouldn't be happening. Segmentation loss is included in the loss function, which forces the mask to match the ground truth mask.

**Comment by pjh4993** [comment]
Hi, Thx for your works.

I also have some question about slight difference in your implementation.

1. In paper, author use both face recogntion model's id and  3dMM's identity, expression, posture vector to construct V_id. However, I found that your implementation only use face recognition model's id. Does this difference was intended? I was also curious about what could go wrong when we drop that from V_id formulation
2. How did you get dilated mask for SFF model's target? I couldn't found any specific guide to get target mask in both paper and Mask preprocessing provided in README.md.
